scripts/install_atos.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+
+# Check if called without arguments
+if [ $# -eq 0 ]; then
+    echo "Don't call this file directly, use setup_atos.sh instead."
+    echo "Usage: ./install_deps.sh <path to ATOS git repo>"           
+    exit 0
+fi
+
+# Take the first argument as the PATH to the ATOS git repo
+ATOS_REPO_PATH="$1"
+source "${ATOS_REPO_PATH}/scripts/install_functions.sh"
+
+################################################
+###### Install Control Panel dependencies ######
+################################################
+
+echo "Installing nvm and Node.js..."
+if [ "$ROS_DISTRO" == "humble" ]; then 
+    NODE_VERSION=16.20.0    # Also specified in the Dockerfile
+elif [ "$ROS_DISTRO" == "foxy" ]; then
+    NODE_VERSION=12
+fi
+NVM_DIR="$HOME/.nvm"
+
+if ! command -v nvm &> /dev/null; then
+    echo "nvm not found, installing..."
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
+    . "$NVM_DIR/nvm.sh"
+    check_command_failed $? "Failed to install nvm."
+else 
+    echo "nvm already installed, skipping installation..."
+fi
+
+echo "Installing Node.js v${NODE_VERSION}..."
+nvm install ${NODE_VERSION}
+nvm use v${NODE_VERSION}
+nvm alias default v${NODE_VERSION}
+echo 'export PATH="$HOME/.nvm/versions/node/v${NODE_VERSION}/bin/:$PATH"' >> $HOME/.bashrc
+source $HOME/.bashrc
+npm --version
+
+# Install pyOpenSSL
+pip install pyOpenSSL
+
+################################################
+############## Install ATOS ####################
+################################################
+
+# Create a workspace dir if it doesn't exist
+if [ ! -d "$HOME/atos_ws/src" ]; then
+    mkdir -p $HOME/atos_ws/src
+fi
+cd $HOME/atos_ws
+
+# Set ATOS_INTERFACES_PATH using ATOS_PATH
+ATOS_INTERFACES_PATH="$ATOS_REPO_PATH/atos_interfaces"
+
+# Update symlinks to atos and atos_interfaces
+update_symlink "$ATOS_REPO_PATH" $HOME/atos_ws/src/atos
+update_symlink "$ATOS_INTERFACES_PATH" $HOME/atos_ws/src/atos_interfaces
+
+
+# Change directory into the workspace and build, check with the user before continuing
+echo "Dependecy installation done and ATOS workspace created."
+
+# First make sure the submodules are up to date
+echo "Updating submodules to make sure they are up to date..."
+cd $HOME/atos_ws/src/atos
+git submodule update --init --recursive
+cd -
+
+# temporarily cd into the workspace and build with colcon
+echo "Building ATOS..."
+cd $HOME/atos_ws
+source /opt/ros/$ROS_DISTRO/setup.bash
+MAKEFLAGS=-j4 && colcon build --symlink-install
```

Comment:
max of j4 and nproc? select number of threads based on output of free -h?

Reply:
"MAKEFLAGS=-jX colcon build" can be used to only set the MAKEFLAGS variable for the next command (and not pollute the user env as "MAKEFLAGS=-jX && colcon build" does)

Reply:
I hade problems running this script in the Dockerfile with -jX, X > 4.